runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -7847,6 +7847,44 @@ J9::X86::TreeEvaluator::VMnewEvaluator(
       {
       genInitObjectHeader(node, clazz, classReg, targetReg, tempReg, monitorSlotIsInitialized, false, cg);
       }
+   TR::Register *discontiguousDataAddrOffsetReg = NULL;
+#ifdef TR_TARGET_64BIT
+   if (isArrayNew)
+      {
+      TR::MemoryReference *dataAddrSlotMR = NULL;
+      TR::MemoryReference *dataAddrMR = NULL;
+      if (TR::Compiler->om.compressObjectReferences() && sizeReg)
+         {
+         if (comp->getOption(TR_TraceCG))
+            traceMsg(comp, "Dealing with compressed refs variable length array. We need to check sizeReg at runtime to determine the correct offset of dataAddr field.\n");
```

Comment:
This seems a bit verbose for a trace message. How about:

```suggestion
         if (comp->getOption(TR_TraceCG))
            traceMsg(comp, "Dealing with compressed refs variable length array.\n");
```

The rest of the message you can put as a comment in the code if you wish.

Reply:
Updated in the new commit.